File: dart/lib/sharedkernel/infrastructure/commandbus/simplecommandbus.dart
import 'dart:collection';
import 'package:dart/sharedkernel/application/command/command.dart';
import 'package:dart/sharedkernel/application/command/handler/handler.dart';
import 'package:dart/sharedkernel/application/commandbus/commandbus.dart';

class SimpleCommandBus implements CommandBus {
  final Map<String, Handler> _handlers;

  SimpleCommandBus() : _handlers = HashMap<String, Handler>();

  @override
  void handle(Command command) {
    _handlers[(command).toString()].handle(command);
  }

  @override
  void registerHandler(Command command, Handler handler) {
    _handlers[(command).toString()] = handler;
  }
}

File: typescript/src/SharedKernel/Infrastructure/CommandBus/SimpleCommandBus.ts
import {Handler} from "../../Application/Command/Handler/Handler";
import {Command} from "../../Application/Command/Command";
import {CommandBus} from "../../Application/CommandBus/CommandBus";

export class SimpleCommandBus implements CommandBus {
    private handlers: { [key: string]: Handler; } = {};

    registerHandler(command: Command, handler: Handler): void {
        this.handlers[command.constructor.toString()] = handler;
    }

    handle(command: Command): void {
        this.handlers[command.constructor.toString()].handle(command);
    }
}